packages/babel-traverse/src/path/conversion.ts

```diff
@@ -397,63 +398,89 @@ function hoistFunctionEnvironment(
   return { thisBinding, fnPath };
 }
 
-function standardizeSuperProperty(superProp) {
+function standardizeSuperProperty(superProp: NodePath<t.MemberExpression>) {
   if (
     superProp.parentPath.isAssignmentExpression() &&
     superProp.parentPath.node.operator !== "="
   ) {
     const assignmentPath = superProp.parentPath;
 
-    const op = assignmentPath.node.operator.slice(0, -1);
+    const op = assignmentPath.node.operator.slice(0, -1) as
+      | LogicalOp
+      | BinaryOp;
+
     const value = assignmentPath.node.right;
 
-    assignmentPath.node.operator = "=";
+    const isLogicalAssignment = op === "||" || op === "&&" || op === "??";
```

Comment:
nit: `@babel/types` exports `LOGICAL_OPERATORS` so we don't have to hardcode them.

Reply:
Updated.
Type narrowing is hard.